### PR TITLE
[REVIEW] Fix DataFrame initialization from list of dicts

### PR DESCRIPTION
**Pull request by brandon-b-miller**
closes https://github.com/rapidsai/cudf/issues/6628

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -7889,3 +7889,22 @@ def test_dataframe_to_pandas_nullable_dtypes(df, expected_pdf):
...
+@pytest.mark.parametrize(
+    "data",
+    [
+        [{"a": 1, "b": 2, "c": 3}, {"a": 4, "b": 5, "c": 6}],
+        [{"a": 1, "b": 2, "c": None}, {"a": None, "b": 5, "c": 6}],
+        [{"a": 1, "b": 2}, {"a": 1, "b": 5, "c": 6}],
+        [{"a": 1, "b": 2}, {"b": 5, "c": 6}],
+        [{}, {"a": 1, "b": 5, "c": 6}],
+        [{"a": 1, "b": 2, "c": 3}, {"a": 4.5, "b": 5.5, "c": 6.5}],
+        [{"a": 1, "b": 2, "c": 3}, {"a": 4.5, "b": 5.5, "c": 6.5}],
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_dataframe.py:7903]: Duplicate test case?

**Review by isVoid — CHANGES_REQUESTED** [review]
minor change requested, otherwise lgtm

File: python/cudf/cudf/core/dataframe.py [hunk 2]
```diff
@@ -398,18 +398,24 @@ def _init_from_list_like(self, data, index=None, columns=None):
...
         self._index = as_index(index)
-        data = list(itertools.zip_longest(*data))
 
-        if columns is not None and len(data) == 0:
-            data = [
-                cudf.core.column.column_empty(row_count=0, dtype=None)
-                for _ in columns
-            ]
+        # list-of-dicts case
+        if len(data) > 0 and np.all(isinstance(d, dict) for d in data):
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:403]:
```suggestion
        if len(data) > 0 and isinstance(data[0], dict):
```
Maybe we can just make it more efficient by just checking the first object?

Comment (brandon-b-miller) [reply python/cudf/cudf/core/dataframe.py:403]: I'm 👍  on that

File: python/cudf/cudf/tests/test_dataframe.py (hunk 1 above)
Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_dataframe.py:7903]: oops! Yep. Removing

**Review by galipremsagar — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
@isVoid does this look ok to you?

**Review by trxcllnt — APPROVED** [review]

**Review by isVoid — APPROVED** [review]
yep! lgtm!

**Pull request merged**